### PR TITLE
searxng: unstable-2023-05-19 -> unstable-2023-06-26

### PR DESCRIPTION
**Pull request by 999eagle**
###### Description of changes

Compare: https://github.com/searxng/searxng/compare/d867bf17e6d2f9a7c83c9a1ffafda5184a24c0e3...da7c30291dcf53cc5b3d98f9aada5615cd1593a9

This update includes the new limiter merged in https://github.com/searxng/searxng/pull/2357. Currently, its configuration file is hardcoded to `/etc/searxng/limiter.toml` (see https://github.com/searxng/searxng/blob/da7c30291dcf53cc5b3d98f9aada5615cd1593a9/searx/botdetection/limiter.py#L73) so the NixOS module for searx would need to write that file if using searxng (maybe as a symlink to `/run/searx/limiter.toml` which is then written by `searx-init.service`?).

The updated package is running with my own searxng instance, deployed with https://git.catgirl.cloud/999eagle/dotfiles-nix/-/commit/95290d72940cb40a7bcb64c9d5a58b9785cdeec7.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/web-apps/searxng/default.nix [hunk 1]
```diff
@@ -50,6 +51,9 @@ python3.pkgs.buildPythonApplication rec {
     # Create a symlink for easier access to static data
     mkdir -p $out/share
     ln -s ../${python3.sitePackages}/searx/static $out/share/
+
+    # copy config schema for the limiter
+    cp searx/botdetection/limiter.toml $out/${python3.sitePackages}/searx/botdetection/limiter.toml
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/searxng/default.nix:56]: Upstream could easily do this with python packaging itself but whatever

**Comment by SuperSandro2000** [comment]
> maybe as a symlink to `/run/searx/limiter.toml` which is then written by `searx-init.service`?

or just use ``environment.etc``.

When we touch the module we should probably also add an option to enable redis and fix the uwsgiConfig for searxng.

**Comment by 999eagle** [comment]
> When we touch the module we should probably also add an option to enable redis and fix the uwsgiConfig for searxng.

Not sure which part of the uwsgiConfig needs fixing, it looks fine to me. Enabling redis would be a good addition though.

As for the limiter, I'm not sure how that would be integrated properly into the NixOS module as the module defaults to searx instead of searxng but the limiter is (as of now) a searxng-specific feature.

**Review by SuperSandro2000 — APPROVED** [review]
tested on my instance and still works as before

**Comment by SuperSandro2000** [comment]
> As for the limiter, I'm not sure how that would be integrated properly into the NixOS module as the module defaults to searx instead of searxng but the limiter is (as of now) a searxng-specific feature.

We should probably switch that over at some point, searx is pretty dead right now.
And we could check if the pname of the package option matches searxng.

**Pull request merged**